packages/ember-data/tests/integration/adapter/json-api-adapter-test.js

```diff
@@ -0,0 +1,831 @@
+var env, store, adapter;
+var passedUrl, passedVerb, passedHash;
+
+var run = Ember.run;
+
+var User, Post, Comment, Handle, GithubHandle, TwitterHandle, Company, DevelopmentShop, DesignStudio;
+
+module('integration/adapter/json-api-adapter - JSONAPIAdapter', {
+  setup: function() {
+    User = DS.Model.extend({
+      firstName: DS.attr('string'),
+      lastName: DS.attr('string'),
+      posts: DS.hasMany('post', { async: true }),
+      handles: DS.hasMany('handle', { async: true, polymorphic: true }),
+      company: DS.belongsTo('company', { async: true, polymorphic: true })
+    });
+
+    Post = DS.Model.extend({
+      title: DS.attr('string'),
+      author: DS.belongsTo('user', { async: true }),
+      comments: DS.hasMany('comment', { async: true })
+    });
+
+    Comment = DS.Model.extend({
+      text: DS.attr('string'),
+      post: DS.belongsTo('post', { async: true })
+    });
+
+    Handle = DS.Model.extend({
+      user: DS.belongsTo('post', { async: true })
+    });
+
+    GithubHandle = Handle.extend({
+      username: DS.attr('string')
+    });
+
+    TwitterHandle = Handle.extend({
+      nickname: DS.attr('string')
+    });
+
+    Company = DS.Model.extend({
+      name: DS.attr('string'),
+      employees: DS.hasMany('user', { async: true })
+    });
+
+    DevelopmentShop = Company.extend({
+      coffee: DS.attr('boolean')
+    });
+
+    DesignStudio = Company.extend({
+      hipsters: DS.attr('number')
+    });
+
+    env = setupStore({
+      adapter: DS.JSONAPIAdapter,
+
+      'user': User,
+      'post': Post,
+      'comment': Comment,
+      'handle': Handle,
+      'github-handle': GithubHandle,
+      'twitter-handle': TwitterHandle,
+      'company': Company,
+      'development-shop': DevelopmentShop,
+      'design-studio': DesignStudio
+    });
+
+    store = env.store;
+    adapter = env.adapter;
+  },
+
+  teardown: function() {
+    run(env.store, 'destroy');
+  }
+});
+
+function ajaxResponse(responses) {
+  var counter = 0;
+  var index;
+
+  passedUrl = [];
+  passedVerb = [];
+  passedHash = [];
+
+  adapter.ajax = function(url, verb, hash) {
+    index = counter++;
+
+    passedUrl[index] = url;
+    passedVerb[index] = verb;
+    passedHash[index] = hash;
+
+    return run(Ember.RSVP, 'resolve', responses[index]);
+  };
+}
+
+if (Ember.FEATURES.isEnabled('ds-new-serializer-api')) {
+
+  test('find a single record', function() {
+    expect(3);
+
+    ajaxResponse([{
+      data: {
+        type: 'post',
+        id: '1',
+        attributes: {
+          title: 'Ember.js rocks'
+        }
+      }
+    }]);
+
+    run(function() {
+      store.find('post', 1).then(function(post) {
+        equal(passedUrl[0], '/post/1');
```

Comment:
@pangratz @wwwdata @SeyZ this PR has now been updated to match the format on jsonapi.org as closely as possible.

- **types** are serialized to, and expected to be **plural dasherized** when received from the server
- **member names** are serialized to, and expected to be **dasherized** when received from the server

However, the internal JSON-API format for ED has **singular dasherized** for types and **camelized** for member names. This should not be an issue as the `JSONAPISerializer` normalizes the format mentioned above to the internal ED format automatically.

All of these will be easily overridden using hooks (`modelNameFromPayloadKey`, `payloadKeyFromModelName`, `extractType`, `keyForAttribute`, `keyForRelationship`...)

URLs are now also built to be **plural dasherized** by default.